### PR TITLE
add lint script & fix code that fails static checking

### PR DESCRIPTION
**Pull request by CoderTH**
Add a script for static checking and fix static check errors in the code

### CODE REVIEW

**Comment by CoderTH** [comment]
@lengrongfu @chaunceyjiang @calvin0327 @wawa0210 cc

I have added ci action for static checking and have been able to work properly, but I have only opened part of the linter. Because of the code history, there are many static checking errors. These bug fixes may affect the existing logic. I suggest that projects such as E2e test have relatively perfect unit tests and open them again after e2e check.

Please help review.

File: hack/verify-staticcheck.sh [hunk 1]
```diff
@@ -0,0 +1,29 @@
...
+cd "${REPO_ROOT}"
+source "hack/util.sh"
+
+if util::cmd_exist golangci-lint ; then
+  echo "Using golangci-lint version:"
+  golangci-lint version
+else
+  echo "Installing golangci-lint ${GOLANGCI_LINT_VER}"
+  # https://golangci-lint.run/usage/install/#other-ci
+  curl -sSfL https://raw.githubusercontent.com/golangci/golangci-lint/master/install.sh | sh -s -- -b $(go env GOPATH)/bin ${GOLANGCI_LINT_VER}
```
Comment (chaunceyjiang) [inline hack/verify-staticcheck.sh]: I don't highly recommend using the code from the `master` branch, you should use a fixed version, such as `v1.57.1` that you defined above.

Comment (CoderTH) [reply hack/verify-staticcheck.sh]: done

File: hack/util.sh [hunk 2]
```diff
@@ -0,0 +1,25 @@
+#!/usr/bin/env bash
+set -o errexit
+set -o nounset
+set -o pipefail
+
+MIN_Go_VERSION=go1.21.0
```
Comment (wawa0210) [inline hack/util.sh]: why "go1.12.0" not directly "1.21.0"

Comment (CoderTH) [reply hack/util.sh]: done

**Pull request merged**